#! /usr/bin/env python

PACKAGE='jsk_perception'

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

gen.add("alpha", double_t, 0, "weight of connectivity energy", 0.1, 0.0, 1.0)
gen.add("beta", double_t, 0, "weight of curvature energy", 0.1, 0.0, 1.0)
gen.add("gamma", double_t, 0, "weight of image energy", 0.1, 0.0, 1.0)
gen.add("window_size", int_t, 0, "window size", 3, 1, 15)
gen.add("max_iterations", int_t, 0, "max iteration", 1000, 1, 10000)
gen.add("epsilon", double_t, 0, "epsilon value of convergence", 0.1, 0.0, 100.0)

exit(gen.generate(PACKAGE, "jsk_perception", "SnakeSegmentation"))
